apps/examples/testcase/le_tc/network/itc_net_connect.c

```diff
@@ -0,0 +1,253 @@
+		return NULL;
+	}
+	while (num_clients-- > 0) {
+		client_socket = accept(server_socket, (struct sockaddr *)&address, (socklen_t*)&addrlen);
+		if (client_socket < 0) {
+			close(server_socket);
+			return NULL;
+		}
+		memset(buffer, 0, BUFF_LEN);
+		valread = read(client_socket, buffer, BUFF_LEN);
+		if (valread < 0) {
+			close(client_socket);
+			close(server_socket);
+			return NULL;
+		}
+		strncpy(buffer + valread, ptr_msg, sizeof(SERVER_MSG));
+		ret = send(client_socket, buffer, strlen(buffer), 0);
+		if (ret != strlen(CLIENT_MSG) + strlen(SERVER_MSG)) {
+			close(client_socket);
+			close(server_socket);
+			return NULL;
+		}
+		sleep(WAIT_TIME_SEC_2);
+		ret = close(client_socket);
+		if (ret != OK) {
+			close(server_socket);
+			return NULL;
+		}
+	}
+	ret = close(server_socket);
+	if (ret != OK) {
+		return NULL;
+	}
+	return NULL;
+}
+
+/**
+* @fn               :client_connect
+* @description      :starts client
+* @return           :void
+*/
+static void *client_connect(void *ptr_id)
+{
+	int *id;
+	id = (int *)ptr_id;
+	int sock = 0;
+	int ret;
+	int valread;
+	struct sockaddr_in serv_addr;
+	char client_msg[BUFF_LEN];
+	char buffer[BUFF_LEN] = {0};
+
+	sock = socket(AF_INET, SOCK_STREAM, 0);
+	if (sock < 0) {
+		return NULL;
+	}
+	memset(&serv_addr, '0', sizeof(serv_addr));
+	serv_addr.sin_family = AF_INET;
+	serv_addr.sin_addr.s_addr = inet_addr("127.0.0.1");
+	serv_addr.sin_port = htons(g_port);
+	ret = connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr));
+	if (ret != OK) {
+		close(sock);
+		return NULL;
+	}
+	strncpy(client_msg, CLIENT_MSG, sizeof(CLIENT_MSG));
+	client_msg[strlen(client_msg) - 2] = '0' + (*id);
+	ret = send(sock, client_msg, strlen(client_msg), 0);
+	if (ret != strlen(client_msg)) {
+		close(sock);
+		return NULL;
+	}
+	valread = read(sock, buffer, BUFF_LEN);
+	if (valread != strlen(client_msg) + strlen(SERVER_MSG)) {
+		close(sock);
+		return NULL;
+	}
+	if (strncmp(buffer, client_msg, strlen(client_msg)) != 0) {
+		close(sock);
+		return NULL;
+	}
+	if (strncmp(buffer + strlen(client_msg), SERVER_MSG, strlen(SERVER_MSG)) != 0) {
+		close(sock);
+		return NULL;
+	}
+	sleep(WAIT_TIME_SEC_2);
+	ret = close(sock);
+	if (ret != OK) {
+		return NULL;
+	}
+	return NULL;
+}
+
+/**
+* @testcase         :itc_net_connect_p
+* @brief            :connects the socket referred to by the file descriptor
+* @scenario         :start server and client thread, connect client to server
+* @apicovered       :connect()
+* @precondition     :none
+* @postcondition    :none
+*/
+static void itc_net_connect_p(void)
+{
+	pthread_t server_thread;
+	pthread_t client_thread;
+	int num_clients = 1;
+	int id = 1;
+	int ret;
+	int *ptr = &num_clients;
+	int *ptr_id = &id;
+	g_port = 8080;
+
+	pthread_create(&server_thread, NULL, server_connect, ptr);
+	sleep(WAIT_TIME_SEC_2);
+	pthread_create(&client_thread, NULL, client_connect, ptr_id);
+
+	sleep(WAIT_TIME_SEC_3);
+
+	ret = pthread_join(server_thread, NULL);
```

Comment:
how do you get result from server_connect()?

Reply:
I can get result by passing "void ** "variable but since test case are using ASSERT which provide Pass/Fail condition; so I didn't validate thread return value.